cmd/krel/cmd/automate.go

```diff
@@ -0,0 +1,223 @@
+	"k8s.io/client-go/util/homedir"
+
+	"k8s.io/release/pkg/automate"
+)
+
+const (
+	k8sOrg string = "kubernetes"
+	websiteRepo string = "website"
+)
+
+type automateOptions struct {
+	githubToken string
+	credFile string
+	sheetId string
+	sheetName string
+	rangeCol string
+	colID []string
+	release string
+}
+
+type PrData struct {
+	KubernetesPR int
+	KEP int
+	WebsitePR int
+	Assignee string
+	Sig string
+}
+
+var automateOpts = automateOptions{
+	credFile:  getCredentialFile(),
+	sheetName: "Docs",
+	rangeCol:  "A1:K",
+	colID: []string{"A", "G", "I", "J", "K"},
+}
+
+// announceCmd represents the subcommand for `krel announce`
+var automateCmd = &cobra.Command{
+	Use:           "automate",
+	Short:         "Automate some work for release docs",
+	SilenceUsage:  true,
+	SilenceErrors: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return cmd.Help()
+	},
+}
+
+func init() {
+	automateCmd.PersistentFlags().StringVarP(
+		&automateOpts.githubToken,
+		"github-token",
+		"",
+		"",
+		"github token for authentication",
+	)
+
+	automateCmd.PersistentFlags().StringVarP(
+		&automateOpts.credFile,
+		"cred-file",
+		"",
+		automateOpts.credFile,
+		"credentials for the Google Sheets API")
+
+	automateCmd.PersistentFlags().StringVarP(
+		&automateOpts.sheetId,
+		"sheet-id",
+		"",
+		"",
+		"Sheet ID of the spreadsheet, normally found in the url of the spreadsheet")
+
+	automateCmd.PersistentFlags().StringVarP(
+		&automateOpts.sheetName,
+		"name",
+		"",
+		automateOpts.sheetName,
+		"The name of the particular sheet that we want to read from")
+
+	automateCmd.PersistentFlags().StringVarP(
+		&automateOpts.rangeCol,
+		"range",
+		"",
+		automateOpts.rangeCol,
+		"The range of columns from which we want to read from e.g A1:B")
+
+	automateCmd.PersistentFlags().StringArrayVar(
+		&automateOpts.colID,
+		"col-ids",
+		automateOpts.colID,
+		// TODO(SomtochiAma): Better explanation
+		"the letters of the columns that contain the kep, k/w, sigs, assignee and k/k pr information")
+
+	automateCmd.PersistentFlags().StringVarP(
+		&automateOpts.release,
+		"release",
+		"",
+		"",
+		"The kubernetes release e.g 1.20")
+
+	rootCmd.AddCommand(automateCmd)
+}
+
+func getPrStruct(sheetID string, readRange string, headers []string) ([]*PrData, error) {
+	if len(headers) == 4 {
+		return nil, fmt.Errorf("column headers should be 5")
+	}
+
+	srv, err := automate.GetSheetService(automateOpts.credFile)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := srv.Spreadsheets.Values.Get(sheetID, readRange).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(resp.Values) == 0 {
+		logrus.Info("No data found")
+		return nil, nil
+	}
+
+
+	headersIdx, err := convertColumnLettersToInt(headers)
+	if err != nil {
+		return nil, err
+	}
+
+	allPrData, err := StructureData(resp.Values, headersIdx)
+	if err != nil {
+		return nil, err
+	}
+
+	return allPrData, nil
+}
+
+func StructureData(data [][]interface{}, index []int) ([]*PrData, error){
```

Comment:
I suggest moving such functions into a dedicated package in `pkg`. This way they can be tested more isolated.